dev/com.ibm.ws.jpa.tests.jpa_32_fat/test-applications/jakartadata/src/io/openliberty/jpa/data/tests/web/JakartaDataRecreateServlet.java

```diff
@@ -1635,6 +1636,41 @@ public void testOLGH29443ZonedDateTime() throws Exception {
         }
     }
 
+   
+    @Test
+   //Reference issue: https://github.com/OpenLiberty/open-liberty/issues/29893
+    public void testOLGH29893() throws Exception {
+        String vehicleId = "V1234";
+        Vehicle vehicle = new Vehicle();
+        vehicle.setId(vehicleId);
+        vehicle.setModel("Toyota Corolla");
+        vehicle.setColor("Blue");
+
+        tx.begin();
+        em.persist(vehicle);
+        tx.commit();
+
+        
+        String idToSearch = "v1234"; 
+        Vehicle result;
+
+        tx.begin();
+        try {
+            result = em.createQuery("FROM Vehicle WHERE LOWER(ID(THIS)) = ?1", Vehicle.class)
+                        .setParameter(1, idToSearch.toLowerCase())
```

Comment:
nitpick: (do not need to change) the variable `idToSearch` is unnecessary.  You could have just used `vehicleId.toLowerCase()`